### ISSUE TITLE
Problem about the nni k8s service

### ISSUE THREAD

**Issue by xieydd** [Microsoft/nni#361]
I deployment the nni as a service in k8s cluster, but when i run a experiment successfully, and i can see the log of trial, it has the accuracy. but the dashboard show all trial accuracy is 0,and the metric in dashboard is NaN.
@QuanluZhang

**Comment by QuanluZhang** [comment]
you have to make the port (default 8080) of nni restserver/webui available outside k8s.

**Comment by xieydd** [comment]
@QuanluZhang Can i change the default port of restserver/webui， if i can , where is the config? Thanks a lot.

**Comment by QuanluZhang** [comment]
please run command `nnictl create --help`, it shows how to set port.

**Comment by xieydd** [comment]
@QuanluZhang  I got it . Thanks for you reply.

**Comment by xieydd** [comment]
@QuanluZhang When i set the same port of nnictl port and service port , but the same problem .

**Comment by yds05** [comment]
Hi, @xieydd , do you use ReplicaSet to launcher NNI pod? If yes, could you please also paset your resource definition file for that ReplicaSet? Thanks.

**Comment by xieydd** [comment]
@yds05  I use the deployment.
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: nni
  labels:
    app: nni
spec:
  replicas: 1   
  template:
    metadata:
      labels:
        app: nni
    spec:
      containers:
      - image: nni:latest
        name : nni
        resources:
          limits:
            nvidia.com/gpu: 1
          requests:
            nvidia.com/gpu: 1
        command: 
          - sleep
          - 360d
        ports:
            containerPort: 30018
```

**Comment by yds05** [comment]
Got it, thanks. The deployment looks good. Another question is which trainingServicePlatform do you set in your config.yml? 

If it's set to local, maybe it's a known issue which  we already fixed in NNI v0.3.4. you can use msranni/nni:v0.3.4 docker image to have a try.

**Comment by xieydd** [comment]
I set it as local, is something wrong?

**Comment by yds05** [comment]
Refer this PR for more detail:
https://github.com/Microsoft/nni/pull/273

Because ts.tail has bug to monitor file change in Docker container, so once you run NNI in docker container, metrics may not be collected correctly without that PR change. 

You can rebuild your docker image based on our latest Dockerfile https://github.com/Microsoft/nni/blob/master/deployment/docker/Dockerfile, or use msranni/nni:v0.3.4  directly.

**Comment by xieydd** [comment]
Thanks  a lot @yds05 , i will test the new image.

**Comment by yds05** [comment]
It looks weird. Could you please share your experiment's log file and sqlite file for us to diagnostic?

The file's location is:
~/nni/experiment/{your_exp_id}/log
~/nni/experiment/{your_exp_id}/db

Thanks.

**Comment by xieydd** [comment]
@yds05 I am sorry, i am holiday these days, i will paste the log tomorrow , thanks a lot .

**Comment by yds05** [comment]
@xieydd , That's fine and thanks for providing these two log files. 
I noticed that your trial config is :

    "trial_config": {
        "gpuNum": 0,
        "codeDir": "/tmp/nni/examples/trials/mnist/.",
        "command": "python3 mnist.py"
    }

Can you make sure the examples folder is built into your docker image? And also, could you please also provide your nni experiment config file? 

Besides that, I didn't find any error directly related to your issue in these log files. And I think I can try to reproduce your issue by staring K8S service to run NNI

**Comment by yds05** [comment]
I see. I will use your config to try to reproduce the issue

**Comment by xieydd** [comment]
Thanks a lot @yds05

**Comment by xieydd** [comment]
@yds05 Have you reproduced the issue?

**Comment by yds05** [comment]
@xieydd  It's weird. I will find another K8S cluster to check.

**Comment by xieydd** [comment]
@yds05 Another question, can you redirect  to the tensorboard?

**Comment by xieydd** [comment]
@yds05 Thanks a lot, i will check my k8s cluster carefully.

**Comment by xieydd** [comment]
[nnimanager.log](https://github.com/Microsoft/nni/files/2618153/nnimanager.log)

@yds05 this is my nnimanager`s log

**Comment by yds05** [comment]
@xieydd Thanks. 

I checked the log, and find indeed there's no metrics sent is recorded in the log. 

Actually, for local mode experiment, we write metrics data into 'metrics' file. 
You can run
'''
kubectl exec -it {your_pod_name} /bin/bash
'''
to enter into your pod. Then goto your trial's working directory, like 
'''
cd /root/nni/experiments/{your_experiment_id}/trials/{your_trial_id}
'''

list the file/directories in trial's working directory.

**Comment by xieydd** [comment]
yep, i can see `metrics` `state` `sequence_id`,but i can`t see the metrics_offset. @yds05 
```
#metrics
ME000112{"sequence": 0, "value": 0.95, "trial_job_id......"}

#state
0 1543299009687

#sequence_id
1
```

**Comment by xieydd** [comment]
![271543299619_ pic](https://user-images.githubusercontent.com/20329697/49062600-a5a21080-f24f-11e8-9c5d-1d1ff8bcd964.jpg)
@yds05

**Comment by xieydd** [comment]
@yds05 You just update your image?
I will try , thanks a lot 👍

**Comment by yds05** [comment]
Yes, I just update my image. Please have a try

**Comment by xieydd** [comment]
@yds05  I can see wodnload button to download the json file, i can see the data.
Haha, Can you give me a progress chart about a stable version ? @yds05

**Comment by yds05** [comment]
@xieydd  Congrats!

We will release NNI v0.4 next week, including pypi package and docker image.

**Comment by xieydd** [comment]
Very Thanks.

**Comment by yds05** [comment]
@xieydd , we released NNI v0.4 yesterday and your issue is fixed in this release. You can use our latest docker image msranni/nni:v0.4 to verify. Thanks.
